src/api/process.c

```diff
@@ -63,12 +76,47 @@ typedef enum {
   REDIRECT_PARENT = -3,
 } filed_e;
 
+static void close_fd(process_handle_t *handle) {
+  if (*handle) {
+#ifdef _WIN32
+    CloseHandle(*handle);
+#else
+    close(*handle);
+#endif
+    *handle = HANDLE_INVALID;
+  }
+}
+
+static int push_error_string(lua_State *L, process_error_t err) {
 #ifdef _WIN32
-  static volatile long PipeSerialNumber;
-  static void close_fd(HANDLE* handle) { if (*handle) CloseHandle(*handle); *handle = INVALID_HANDLE_VALUE; }
+  char *msg = NULL;
+  FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM
+                | FORMAT_MESSAGE_ALLOCATE_BUFFER
+                | FORMAT_MESSAGE_IGNORE_INSERTS,
+                NULL,
+                err,
+                MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                (char *) &msg,
+                0,
+                NULL);
+  if (!msg)
+    return 0;
+
+  lua_pushstring(L, msg);
+  LocalFree(msg);
 #else
-  static void close_fd(int* fd) { if (*fd) close(*fd); *fd = 0; }
+  lua_pushstring(L, strerror(err));
 #endif
+  return 1;
+}
+
+static void push_error(lua_State *L, const char *extra, process_error_t err) {
```

Comment:
`const char *extra` doesn't seems to be used anywhere on the new code and instead set to NULL everywhere, are you planning to use it on the future?

Reply:
It was planned for the future to print something like "error starting process: permission denied (1)"

Reply:
would be nice if the old error hints are kept then, instead of just doing "error: {system_error_message}"